src/reactapp/src/components/shippingAddress/components/ShippingAddressFormikProvider.jsx

```diff
@@ -95,19 +98,28 @@ function ShippingAddressFormikProvider({ children, formikData }) {
   const cartHasShippingAddress = isCartAddressValid(cartShippingAddress);
 
   const resetShippingAddressFormFields = useCallback(() => {
-    setFieldValue(SHIPPING_ADDR_FORM, { ...initialValues });
+    setFieldValue(SHIPPING_ADDR_FORM, { ...defaultValues });
     setFieldTouched(SHIPPING_ADDR_FORM, {});
   }, [setFieldValue, setFieldTouched]);
 
   const setShippingAddressFormFields = useCallback(
     (addressToSet) =>
       setFieldValue(SHIPPING_ADDR_FORM, {
-        ...initialValues,
+        ...defaultValues,
         ...addressToSet,
       }),
     [setFieldValue]
   );
 
+  // Update initialvalues based on the initial cart data fetch.
+  useEffect(() => {
+    if (aggregatedData) {
+      const shippingAddress = aggregatedData?.cart?.shipping_address || {};
+      const saveInBook = !!aggregatedData?.customer?.customer?.email;
+      setInitialValues({ ...defaultValues, ...shippingAddress, saveInBook });
+    }
```

Comment:
Personally I would prefer an early return:

`if (!aggregatedData) return`

Reply:
i also like the approach, except in `useEffect`. Because `return` in `useEffect` has different meaning. We can return a function for example in order to do some actions during component unmount. Due to that reason, I normally give an exception for useeffect if it contains only a single condition. If more than sing condition present, then an early return makes sense there. However, simpler the side effect content, better it is :)

Reply:
Ah, of course! I already thought there was a reason for not doing this here. So, this one is resolved 😄